### PR TITLE
Expand validation and status spec for deployment hooks

### PR DESCRIPTION
**Pull request by ironcladlou**
The deployment hook proposal needs some amendments:

1. Cover validations
2. Cover the relationship between hook status and deployment status

### CODE REVIEW

**Comment by ironcladlou** [comment]
cc @smarterclayton @bparees

File: docs/proposals/post-deployment-hooks.md [hunk 1]
```diff
@@ -250,6 +250,22 @@ const (
 )
 ```
 
+#### Validations related to hooks
+
+Initially, valid values for `Lifecycle.Post.FailurePolicy` will be `Retry` and `Continue`. This may change in the future if deployments can be safely rolled back automatically.
```
Comment (bparees) [inline docs/proposals/post-deployment-hooks.md]: Would "Ignore" make more sense than "Continue"?

Comment (ironcladlou) [inline docs/proposals/post-deployment-hooks.md]: >Would "Ignore" make more sense than "Continue"?
Hmmm, I think so.

**Comment by smarterclayton** [comment]
LGTM [merge]

**Comment by smarterclayton** [comment]
[merge]

**Pull request merged**